File: solutions/c/two-fer/1/two_fer.c
#include "two_fer.h"
#include <stdio.h>
#include <string.h>

void two_fer(char *buffer, const char *name) {
    if (name == NULL || strlen(name) == 0) {
        // No name provided
        sprintf(buffer, "One for you, one for me.");
    } else {
        // Name provided
        sprintf(buffer, "One for %s, one for me.", name);
    }
}

File: solutions/go/two-fer/1/two_fer.go
// Package twofer provides a function to share a cookie.
package twofer

import "fmt"

// ShareWith returns a string saying "One for X, one for me."
// If the name is empty, X is "you".
func ShareWith(name string) string {
	if name == "" {
		name = "you"
	}
	return fmt.Sprintf("One for %s, one for me.", name)
}